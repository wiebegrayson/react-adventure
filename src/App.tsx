import { useEffect, useState } from "react";
import Card, { IProps as IData } from "./components/Card";
import { Wrap, WrapItem } from "@chakra-ui/react";
import { createApi } from "unsplash-js";

const unsplash = createApi({
  accessKey: import.meta.env.VITE_UNSPLASH_ACCESS_KEY
});

export const getImgForDataEntry = async (
  entry: Omit<IData, "img">
): Promise<IData> => {
  return unsplash.search
    .getPhotos({
      query: `${entry.title}`,
      perPage: 1
    })
    .then(({ response }) => ({
      ...entry,
      img: response?.results[0].urls.regular || ""
    }));
};

function App() {
  const [mockData, setMockData] = useState<IData[] | null>(null);

  useEffect(() => {
    const dataURL = new URL("./assets/data.json", import.meta.url).href;
    fetch(dataURL)
      .then(res => res.json())
      .then(async (data: Omit<IData, "img">[]) => {
        const newData = await Promise.all(data.map(getImgForDataEntry));
        setMockData(newData);
      });
  }, []);

  return (
    <Wrap
      minHeight="100vh"
      display="grid"
      placeItems="center"
      align="center"
      sx={{ "& ul": { width: "100%" } }}
      justify="center"
      bgColor="#ddd"
      spacingY={4}
      spacingX={8}
    >
      {mockData?.map(data => (
        <WrapItem sx={{ width: "min(465px, 100%)" }} key={data.title}>
          <Card {...data} />
        </WrapItem>
      ))}
    </Wrap>
  );
}

export default App;
