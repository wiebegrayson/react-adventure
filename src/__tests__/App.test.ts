import { expect } from "vitest";
import { getImgForDataEntry } from "../App";

test("Get image for data entry", async () => {
  const dataEntry = {
    title: "European Quest",
    subtitle: "8 countries, 21 days",
    emissions: "810 kg",
    rating: 4.7
  };

  const result = await getImgForDataEntry(dataEntry);

  expect(result).toHaveProperty("img");
  expect(result.img).toBeTruthy();
  expect(result.img).toBeTypeOf("string");
});
