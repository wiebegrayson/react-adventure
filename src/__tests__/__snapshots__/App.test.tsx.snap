// Vitest Snapshot v1

exports[`Card component renders mock data 1`] = `
<div
  className="css-0"
>
  <h2
    className="chakra-text css-0"
  >
    Autumn Roadtrip
  </h2>
  <p
    className="chakra-text css-0"
  >
    1 country, 14 days
  </p>
  <div
    className="chakra-stack css-0"
  >
    <p
      className="chakra-text css-0"
    >
      Emissions offset:
    </p>
    <p
      className="chakra-text css-0"
    >
      3.56 t
       CO
      <sub>
        2
      </sub>
      e
    </p>
  </div>
  <div
    className="chakra-stack css-0"
  >
    <p
      className="chakra-text css-0"
    >
      Trip rating
    </p>
    <div
      className="chakra-stack css-0"
    >
      <div
        className="chakra-stack css-0"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </div>
      <p
        className="chakra-text css-0"
      >
        4.7
      </p>
    </div>
  </div>
</div>
`;

exports[`Card component renders mock data 2`] = `
<div
  className="css-0"
>
  <h2
    className="chakra-text css-0"
  >
    Autumn Roadtrip
  </h2>
  <p
    className="chakra-text css-0"
  >
    1 country, 14 days
  </p>
  <div
    className="chakra-stack css-0"
  >
    <p
      className="chakra-text css-0"
    >
      Emissions offset:
    </p>
    <p
      className="chakra-text css-0"
    >
      3.56 t
       CO
      <sub>
        2
      </sub>
      e
    </p>
  </div>
  <div
    className="chakra-stack css-0"
  >
    <p
      className="chakra-text css-0"
    >
      Trip rating
    </p>
    <div
      className="chakra-stack css-0"
    >
      <div
        className="chakra-stack css-0"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </div>
      <p
        className="chakra-text css-0"
      >
        4.7
      </p>
    </div>
  </div>
</div>
`;

exports[`Link changes the class when hovered 1`] = `
<div
  className="css-0"
>
  <h2
    className="chakra-text css-0"
  >
    Autumn Roadtrip
  </h2>
  <p
    className="chakra-text css-0"
  >
    1 country, 14 days
  </p>
  <div
    className="chakra-stack css-0"
  >
    <p
      className="chakra-text css-0"
    >
      Emissions offset:
    </p>
    <p
      className="chakra-text css-0"
    >
      3.56 t
       CO
      <sub>
        2
      </sub>
      e
    </p>
  </div>
  <div
    className="chakra-stack css-0"
  >
    <p
      className="chakra-text css-0"
    >
      Trip rating
    </p>
    <div
      className="chakra-stack css-0"
    >
      <div
        className="chakra-stack css-0"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </div>
      <p
        className="chakra-text css-0"
      >
        4.7
      </p>
    </div>
  </div>
</div>
`;

exports[`Link changes the class when hovered 2`] = `
<div
  className="css-0"
>
  <h2
    className="chakra-text css-0"
  >
    Autumn Roadtrip
  </h2>
  <p
    className="chakra-text css-0"
  >
    1 country, 14 days
  </p>
  <div
    className="chakra-stack css-0"
  >
    <p
      className="chakra-text css-0"
    >
      Emissions offset:
    </p>
    <p
      className="chakra-text css-0"
    >
      3.56 t
       CO
      <sub>
        2
      </sub>
      e
    </p>
  </div>
  <div
    className="chakra-stack css-0"
  >
    <p
      className="chakra-text css-0"
    >
      Trip rating
    </p>
    <div
      className="chakra-stack css-0"
    >
      <div
        className="chakra-stack css-0"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </div>
      <p
        className="chakra-text css-0"
      >
        4.7
      </p>
    </div>
  </div>
</div>
`;

exports[`Link changes the class when hovered 3`] = `
<div
  className="css-0"
>
  <h2
    className="chakra-text css-0"
  >
    Autumn Roadtrip
  </h2>
  <p
    className="chakra-text css-0"
  >
    1 country, 14 days
  </p>
  <div
    className="chakra-stack css-0"
  >
    <p
      className="chakra-text css-0"
    >
      Emissions offset:
    </p>
    <p
      className="chakra-text css-0"
    >
      3.56 t
       CO
      <sub>
        2
      </sub>
      e
    </p>
  </div>
  <div
    className="chakra-stack css-0"
  >
    <p
      className="chakra-text css-0"
    >
      Trip rating
    </p>
    <div
      className="chakra-stack css-0"
    >
      <div
        className="chakra-stack css-0"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </div>
      <p
        className="chakra-text css-0"
      >
        4.7
      </p>
    </div>
  </div>
</div>
`;
